adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -815,25 +820,29 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] extends Logging {
    * in the product of the join.
    *
    * @param genomicRdd The right RDD in the join.
+   * @param threshold Sets a threshold for the distance between elements to be
+   *   joined. If set to 0, an overlap is required to join two elements.
    * @return Returns a new genomic RDD containing all pairs of keys that
    *   overlapped in the genomic coordinate space, and all keys from the
    *   right RDD that did not overlap a key in the left RDD.
    *
    * @see rightOuterBroadcastRegionJoin
    */
   def rightOuterBroadcastRegionJoin[X, Y <: GenomicRDD[X, Y], Z <: GenomicRDD[(Option[T], X), Z]](
-    genomicRdd: GenomicRDD[X, Y])(
+    genomicRdd: GenomicRDD[X, Y],
+    threshold: Long = 0L)(
       implicit tTag: ClassTag[T],
       xTag: ClassTag[X],
       otxTag: ClassTag[(Option[T], X)]): GenomicRDD[(Option[T], X), Z] = RightOuterBroadcastJoin.time {
 
     // key the RDDs and join
     GenericGenomicRDD[(Option[T], X)](RightOuterTreeRegionJoin[T, X]().broadcastAndJoin(
-      buildTree(flattenRddByRegions()),
+      buildTree(flattenRddByRegions().map(f => (f._1.pad(threshold), f._2))),
       genomicRdd.flattenRddByRegions()),
       sequences ++ genomicRdd.sequences,
       kv => {
-        Seq(kv._1.map(v => getReferenceRegions(v))).flatten.flatten ++
+        Seq(kv._1.map(v => getReferenceRegions(v)
+          .map(_.pad(-1 * threshold)))).flatten.flatten ++
```

Comment:
I did a double-take on the lines that pad by a negative threshold. Can you add documentation to explain that we are removing the padding that we added earlier?

Reply:
Alternatively we could add an `unpad` and put the documentation there. It doesn't have to be public, but it would make this cleaner.

Reply:
I think a `//` style single line comment would be sufficient